app/src/ui/dialog/dialog.tsx

```diff
@@ -104,6 +106,15 @@ interface IDialogProps {
    * of the loading operation.
    */
   readonly loading?: boolean
+
+  /**
+   * An optional max width for the dialog.
+   * This is used to make a dialog width's responsive.
+   *
+   * On window resize, the dialog will adjust to fill the app window width sans
+   * padding within the constrains of the this max width and a css defined min.
+   */
+  readonly maxWidth?: number
```

Comment:
I originally had an IContrainValue, but determined it was useless. :P We need to have a minimum as low as the css set minimum of 400 to support being able to close a dialog when zoomed in all the way. Therefore, the maxWidth is the only determining factor.